service/history/workflow/cache/cache.go

```diff
@@ -462,3 +487,10 @@ func makeCacheKey(
 		ShardUUID:   shardContext.GetOwner(),
 	}
 }
+
+func (c *cacheItem) CacheSize() int {
+	if sg, ok := c.wfContext.(cache.SizeGetter); ok {
+		return sg.CacheSize()
+	}
+	return 0
+}
```

Comment:
Redirects the cache size request to the inner workflow context.

Note that the cast is needed since some tests pass a MockContext in that does not implement this interface.